src/coreclr/jit/compiler.cpp

```diff
@@ -2892,9 +2894,22 @@ void Compiler::compInitOptions(JitFlags* jitFlags)
             fgNumProfileRuns = 0;
             for (UINT32 iSchema = 0; iSchema < fgPgoSchemaCount; iSchema++)
             {
-                if (fgPgoSchema[iSchema].InstrumentationKind == ICorJitInfo::PgoInstrumentationKind::NumRuns)
+                switch (fgPgoSchema[iSchema].InstrumentationKind)
                 {
-                    fgNumProfileRuns += fgPgoSchema[iSchema].Other;
+                    case ICorJitInfo::PgoInstrumentationKind::NumRuns:
```

Comment:
compInitOptions seems like the wrong place for all this profile code. Can't is be done in the caller (compCompileHelper)? You just want it done before fgFindBasicBlocks? Maybe it would be useful to leave a comment here about the ordering requirements?

Reply:
Good point, I think it can all move into `fgIncorporateProfileData`.

Reply:
At least the analysis bit can move -- the VM request probably needs to stay there.

Reply:
Just to elaborate -- currently there is a thin dependence on checking availability of profile data early like this, basically we have dump output that describes what is happening.

But longer-term we might actually want to alter jit behavior if asked to do BBOPT but no profile data is available, so it seems reasonable (for now) to leave this bit of logic where it is.